FreeAPS/Sources/Services/WatchManager/WatchManager.swift

```diff
@@ -216,9 +217,11 @@ extension BaseWatchManager: WCSessionDelegate {
         debug(.service, "WCSession got message with reply handler: \(message)")
 
         if let carbs = message["carbs"] as? Double, carbs > 0 {
-            carbsStorage.storeCarbs([
-                CarbsEntry(createdAt: Date(), carbs: Decimal(carbs), enteredBy: CarbsEntry.manual)
-            ])
+            let carbArray = [
+                CarbsEntry(id: UUID().uuidString, createdAt: Date(), carbs: Decimal(carbs), enteredBy: CarbsEntry.manual)
+            ]
+            carbsStorage.storeCarbs(carbArray)
+            healthKitManager.saveIfNeeded(carbs: carbArray)
```

Comment:
Please read comment, where I tell about `CarbObserver`. 
After that these edits are unnecessary